scripts/deploy.md

```diff
@@ -0,0 +1,388 @@
+
+[ 2/23] Check that git doesn't have uncommited changes...
+$ git diff-index --quiet HEAD --
+
+[ 3/23] Fetch fresh version from upstream...
+$ git fetch
+
+[ 4/23] Checkout master...
+$ git checkout master
+Already on 'master'
+Your branch is up to date with 'origin/master'.
+
+[ 5/23] Compare SHA of /home/crem/dev/test/scripts/deploy.py...
+SHA256 of /home/crem/dev/test/scripts/deploy.py is c04e3279dedcc9b2a5526a1eca3914e6a6529c266507546a42e0f0a135825883
+
+[ 6/23] Checkout release...
+$ git checkout release
+Switched to branch 'release'
+Your branch is up to date with 'origin/release'.
+
+[ 7/23] Read current version from version.inc file...
+Found current version v1.3.1
+
+[ 8/23] Select new major/minor version....
+Current version is v1.3.1. What will be the new one?
+  1. 1.4-rc1
+  2. 2.0-rc1
+>>>>>>> 1
+
+[ 9/23] Merge master into release...
+$ git merge master
+Already up to date.
+
+[10/23] Store git log for appending to changelog.txt...
+
+[11/23] Prepend git log to a changelog...
+
+[12/23] Store SHA of /home/crem/dev/test/changelog.txt...
+SHA256 of /home/crem/dev/test/changelog.txt is 7e77a949b543dce2fa324d047599ab0927107bc9a92b6077a8b0f52ff3f55d37
+
+[13/23] Edit changelog...
+$ vim /home/crem/dev/test/changelog.txt
+
+[14/23] Compare SHA of /home/crem/dev/test/changelog.txt...
+SHA256 of /home/crem/dev/test/changelog.txt is 5cb0aab523f19b7387a362a145e29983386399b50a7cffcdad1dbf8b981c736d
+
+[15/23] Execution of git add /home/crem/dev/test/changelog.txt ...
+$ git add /home/crem/dev/test/changelog.txt
+
+[16/23] Write version into version.inc, commit and tag....
+$ git add /home/crem/dev/test/src/version.inc
+$ git commit -m "Change version.inc to v1.5-dev."
+[release 70e12c6] Change version.inc to v1.5-dev.
+ 2 files changed, 8 insertions(+), 3 deletions(-)
+
+[17/23] Checkout release...
+$ git checkout master
+Switched to branch 'master'
+Your branch is up to date with 'origin/master'.
+
+[18/23] Merge release into master...
+$ git merge release
+Updating 232136d..70e12c6
+Fast-forward
+ changelog.txt   | 25 +++++++++++++++++++++++++
+ src/version.inc |  2 +-
+ 2 files changed, 26 insertions(+), 1 deletion(-)
+
+[19/23] Checkout release...
+$ git checkout release
+Switched to branch 'release'
+Your branch is ahead of 'origin/release' by 1 commit.
+  (use "git push" to publish your local commits)
+
+[20/23] Write version into version.inc, commit and tag....
+$ git add /home/crem/dev/test/src/version.inc
+$ git commit -m "Change version.inc to v1.4-rc1."
+[release 2b515ea] Change version.inc to v1.4-rc1.
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+$ git tag -a v1.4-rc1 -m "Adding tag v1.4-rc1"
+
+[21/23] Checkout master...
+$ git checkout master
+Switched to branch 'master'
+Your branch is ahead of 'origin/master' by 6 commits.
+  (use "git push" to publish your local commits)
+
+[22/23] Push release changes to github...
+$ git push --all origin
```

Comment:
How does this work?

As our repo is currently designed, no commits may be pushed to master without an approving review, i.e. all pushes must have a PR with input from another human.

Reply:
I'm pretty sure all of those branch protections are only in github UI, and through command line it's all unrestricted, at least to repo admins.

Reply:
That is not true (as mentioned on Discord, but worth duplicating here for those not on Discord). The branch protections apply to all means of pushing, GH UI or git CLI.

Reply:
Yes indeed, I've checked and it doesn't allow me to do that. Will need to find a solution.